gammapy/datasets/core.py

```diff
@@ -79,7 +79,10 @@ def _stat_sum_likelihood(self):
         stat = self.stat_array()
 
         if self.mask is not None:
-            stat = stat[self.mask.data]
+            if isinstance(self.mask, np.ndarray):
```

Comment:
Maybe it is better to have the mask as a Map always. I remember there was some confusion when we shifted from ndarray to map